core/src/main/java/org/jdeferred/impl/AbstractPromise.java

```diff
@@ -128,33 +133,56 @@ protected void triggerProgress(P progress) {
 				triggerProgress(callback, progress);
 			} catch (Exception e) {
 				log.error("an uncaught exception occured in a ProgressCallback", e);
+                saveExceptionIfRuntimeException(e);
+                for (ExceptCallback exceptCallback : exceptCallbacks) {
+                    triggerExcept(exceptCallback, e);
+                }
 			}
 		}
 	}
-	
+
+
 	protected void triggerProgress(ProgressCallback<P> callback, P progress) {
 		callback.onProgress(progress);
 	}
-	
+
 	protected void triggerAlways(State state, D resolve, F reject) {
 		for (AlwaysCallback<D, F> callback : alwaysCallbacks) {
 			try {
 				triggerAlways(callback, state, resolve, reject);
 			} catch (Exception e) {
 				log.error("an uncaught exception occured in a AlwaysCallback", e);
+                saveExceptionIfRuntimeException(e);
+                for (ExceptCallback exceptCallback : exceptCallbacks) {
+                    triggerExcept(exceptCallback, e);
+                }
 			}
 		}
 		alwaysCallbacks.clear();
-		
+
 		synchronized (this) {
 			this.notifyAll();
 		}
 	}
-	
+
 	protected void triggerAlways(AlwaysCallback<D, F> callback, State state, D resolve, F reject) {
 		callback.onAlways(state, resolve, reject);
 	}
 
+    private void saveExceptionIfRuntimeException(Exception e) {
+        if (e instanceof RuntimeException) {
+            thrownRuntimeException = (RuntimeException) e;
+        }
+    }
+
+    private void triggerExcept(ExceptCallback exceptCallback, Exception e) {
+        try {
+            exceptCallback.onException(e);
+        } catch (Exception ex) {
+            log.error("an uncaught exception occured in an ExceptCallback", e);
```

Comment:
Exception *ex* occurs on the calling code, but the internals of the Promise gobbles it up. Would it be a good idea to let the exception bubble up? That way an `UncaughtExceptionHandler` may be able to handle this exception.